app.rb

```diff
@@ -0,0 +1,125 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No books available.'
+    else
+      puts 'All Books:'
+      @books.each do |book|
+        puts "Title: #{book.title} || Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'No people available.'
+    else
+      puts 'All People:'
+      @people.each do |person|
+        puts "ID: #{person.id} || Name: #{person.name} || Type: #{person.class}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a Student(1) or a Teacher(2)? [Input the number]: '
+    res = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    if res == 1
+      person = Student.new(age, nil, name)
+      puts "Student created successfully. ID: #{person.id}"
+      @people << person
+    elsif res == 2
+      print 'Enter the teacher\'s specialization: '
+      specialization = gets.chomp
+      person = Teacher.new(age, specialization, name)
+      puts "Teacher created successfully. ID: #{person.id}"
+      @people << person
+    else
+      puts 'Invalid choice. Person creation failed.'
+    end
+  end
+
+  def create_book
+    print 'Enter the book\'s title: '
+    title = gets.chomp
+    print 'Enter the book\'s author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def select_person
+    puts 'Select a person to create a rental:'
+    list_all_people
+    print 'Enter the person\'s ID: '
+    person_id = gets.chomp.to_i
+    @people.find { |p| p.id == person_id }
+  end
+
+  def create_rental
+    person = select_person
+    puts 'Person not found.' and return if person.nil?
+
+    puts 'Select a book to rent:'
+    list_all_books
+    print 'Enter the book\'s title: '
```

Comment:
- [ ] I think it would be a better user experience if you ask the user to select a book using the book's index instead of the title. Please update this functionality according to the sample video. ✍️